src/openforms/analytics_tools/models.py

```diff
@@ -0,0 +1,312 @@
+        if original_object.enable_google_analytics != self.enable_google_analytics or (
+            not self.google_analytics_can_be_enabled() and self.enable_google_analytics
+        ):
+            if not self.google_analytics_can_be_enabled():
+                self.enable_google_analytics = False
+            self.update_analytics_tool(
+                AnalyticsTools.google_analytics, self.google_analytics_can_be_enabled()
+            )
+        if (
+            original_object.enable_matomo_site_analytics
+            != self.enable_matomo_site_analytics
+            or (not self.matomo_can_be_enabled() and self.enable_matomo_site_analytics)
+        ):
+            if not self.matomo_can_be_enabled():
+                self.enable_matomo_site_analytics = False
+            self.update_analytics_tool(
+                AnalyticsTools.matomo,
+                self.matomo_can_be_enabled(),
+                self.matomo_site_id,
+                self.matomo_url,
+            )
+        if (
+            original_object.enable_piwik_pro_site_analytics
+            != self.enable_piwik_pro_site_analytics
+            or (
+                not self.piwik_pro_can_be_enabled()
+                and self.enable_piwik_pro_site_analytics
+            )
+        ):
+            if not self.piwik_pro_can_be_enabled():
+                self.enable_piwik_pro_site_analytics = False
+            self.update_analytics_tool(
+                AnalyticsTools.piwik_pro,
+                self.piwik_pro_can_be_enabled(),
+                self.piwik_pro_site_id,
+                self.piwik_pro_url,
+            )
+        if (
+            original_object.enable_piwik_site_analytics
+            != self.enable_piwik_site_analytics
+            or (not self.piwik_can_be_enabled() and self.enable_piwik_site_analytics)
+        ):
+            if not self.piwik_can_be_enabled():
+                self.enable_piwik_site_analytics = False
+            self.update_analytics_tool(
+                AnalyticsTools.piwik,
+                self.piwik_can_be_enabled(),
+                self.piwik_site_id,
+                self.piwik_url,
+            )
+        if (
+            original_object.enable_siteimprove_analytics
+            != self.enable_siteimprove_analytics
+            or (
+                not self.siteimprove_can_be_enabled()
+                and self.enable_siteimprove_analytics
+            )
+        ):
+            if not self.siteimprove_can_be_enabled():
+                self.enable_siteimprove_analytics = False
+            self.update_analytics_tool(
+                AnalyticsTools.siteimprove, self.siteimprove_can_be_enabled()
+            )
+        super().save(*args, **kwargs)
+
+    def update_analytics_tool(
+        self,
+        analytics_tool: str,
+        is_activated: bool,
+        site_id: Optional[str] = None,
+        site_url: Optional[str] = None,
```

Comment:
these are quite tool-specific arguments in a generic method. Maybe it would be better suited to pass in a list of string replacements? e.g. `[("SITE_ID", site_id), ("SITE_URL", site_url)]`